application/src/main/java/org/togetherjava/tjbot/commands/moderation/attachment/AttachmentListener.java

```diff
@@ -0,0 +1,118 @@
+        if (event.getAuthor().isBot() || event.isWebhookMessage()) {
+            return;
+        }
+        if (attachmentContainsBlacklistedFileExtension(event.getMessage())) {
+            deleteMessageAndDmUser(event);
+            warnMods(event);
+        }
+    }
+
+    private void deleteMessageAndDmUser(@NotNull MessageReceivedEvent event) {
+        event.getMessage().delete().flatMap(any -> dmUser(event)).queue();
+    }
+
+    private RestAction<Message> dmUser(@NotNull MessageReceivedEvent event) {
+        Message dmMessage = createDmMessage(event.getMessage());
+        return event.getJDA()
+            .openPrivateChannelById(event.getAuthor().getIdLong())
+            .flatMap(channel -> channel.sendMessage(dmMessage));
+    }
+
+    private Message createDmMessage(Message originalMessage) {
+        String originalMessageContent = originalMessage.getContentDisplay();
+        String blacklistedAttachments = getBlacklistedAttachmentsFromMessage(originalMessage);
+        String dmMessageContent =
+                "Hey there, you posted a message with a blacklisted file attachment: %s. Following file extension are blacklisted: %s ."
+                    .formatted(blacklistedAttachments, config.getBlacklistedFileExtensions());
+        // No embedded needed if there was no message from the user
+        if (StringUtils.isEmpty(originalMessageContent)) {
+            return new MessageBuilder(dmMessageContent).build();
+        }
+        return createMessageWithOriginalMessageAsEmbedded(originalMessageContent, dmMessageContent);
+    }
+
+    private Message createMessageWithOriginalMessageAsEmbedded(String originalMessageContent,
+            String dmMessageContent) {
+        MessageEmbed originalMessageEmbed =
+                new EmbedBuilder().setDescription(originalMessageContent)
+                    .setColor(Color.ORANGE)
+                    .build();
+        return new MessageBuilder(dmMessageContent).setEmbeds(originalMessageEmbed).build();
+    }
+
+    private String getBlacklistedAttachmentsFromMessage(Message originalMessage) {
+        List<String> blacklistedAttachments = new ArrayList<>();
+        originalMessage.getAttachments()
+            .stream()
+            .filter(attachment -> config.getBlacklistedFileExtensions()
+                .contains(attachment.getFileExtension()))
+            .forEach(attachment -> blacklistedAttachments.add(attachment.getFileName()));
+        return String.join(", ", blacklistedAttachments);
+    }
+
+    private boolean attachmentContainsBlacklistedFileExtension(Message message) {
+        List<Message.Attachment> attachments = message.getAttachments();
+        if (attachments.isEmpty()) {
+            return false;
+        }
+        return attachments.stream()
+            .anyMatch(attachment -> config.getBlacklistedFileExtensions()
+                .contains(attachment.getFileExtension()));
```

Comment:
ur doing these `contains` on the config very often. at that point, it would be worth to just put the list into a `Set<String>` as a field in ur class and then use `Set#contains`, much faster. And u dont have to constantly ask the config.